### ISSUE TITLE
AttributeError: '_PyDrive2ImportHook' object has no attribute 'find_spec'

### ISSUE THREAD

**Issue by jslegers** [python/cpython#122665]
# Bug report

### Bug description:

I got an error when trying to run the following code from https://docs.python.org/3/library/importlib.html#importlib.util.spec_from_loader on Google Colab :

````python
def import_module(name, package=None):
    """An approximate implementation of import."""
    absolute_name = importlib.util.resolve_name(name, package)
    try:
        return sys.modules[absolute_name]
    except KeyError:
        pass

    path = None
    if '.' in absolute_name:
        parent_name, _, child_name = absolute_name.rpartition('.')
        parent_module = import_module(parent_name)
        path = parent_module.__spec__.submodule_search_locations
    for finder in sys.meta_path:
        spec = finder.find_spec(absolute_name, path)
        if spec is not None:
            break
    else:
        msg = f'No module named {absolute_name!r}'
        raise ModuleNotFoundError(msg, name=absolute_name)
    module = importlib.util.module_from_spec(spec)
    sys.modules[absolute_name] = module
    spec.loader.exec_module(module)
    if path is not None:
        setattr(parent_module, child_name, module)
    return module
````

This is the error I get :
````python
AttributeError: '_PyDrive2ImportHook' object has no attribute 'find_spec'
````

The following line triggered the error : 

````python
spec = finder.find_spec(absolute_name, path)
````

I have a similar issue when I run the following code : 

````python
def load_module(module_name : str) -> ModuleType :
  try :
    if module_name in sys.modules:
      print(f"{module_name} already in sys.modules")
      return sys.modules[module_name]
    if (spec := importlib.util.find_spec(module_name)) is not None :
      module = importlib.util.module_from_spec(spec)
      spec.loader.exec_module(module)
      return module
  except Exception as e :
    print(e)
````

--------------

This works fine for pretty much ever module I've tested :
````python
module = importlib.import_module(name)
````

For certain edge cases, I use this as a workaround :
````python
module = importlib.import_module(name)
spec = module.__spec__
spec.loader.exec_module(module)
````

This also works fine for me :
````python
source_code = f"from {module_name} import {name}"
module = import_from_string(module_name+'.'+name, source_code)
sys.modules[self.__name__ + '.' + name] = module
````

### CPython versions tested on:

3.10

### Operating systems tested on:

Linux

**Comment by aisk** [comment]
The `_PyDrive2ImportHook` is not part of Python, I guess it refers to some Google Drive libraries, so I think this issue is not related to Python.

**Comment by terryjreedy** [comment]
OP, what did you actually do?  `following code from ...` links to doc entry for function `spec_from_loader`.  There is no code there, and your `def import_module` statement does not use that statement.  When that single statement code is run, it compiles fine to a function object named 'import_module'.  If you called that function, what were the arguments?  As aisk implies, one should test this sort of code directly with python rather than a 3rd-party environment.  When people run apparently valid code in IDLE and think it reveals a system bug, and I agree, I tell them to rerun directly in Python, without IDLE, to eliminate IDLE as the possible cause.  Also, we don't fix non-security bugs before 3.12, so they must exist in 3.12+.
